File: backend/app/schemas/eval.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class EvalRunRequest(BaseModel):
    strategy: str = "hybrid"  # "hybrid" or "dense_only"
    reranking_enabled: bool = True


class EvalResultResponse(BaseModel):
    id: uuid.UUID
    run_id: str
    strategy: str
    reranking_enabled: bool
    faithfulness: float | None
    answer_relevancy: float | None
    context_precision: float | None
    context_recall: float | None
    per_question_results: dict | None
    created_at: datetime

    model_config = {"from_attributes": True}


class EvalResultsListResponse(BaseModel):
    results: list[EvalResultResponse]
    total: int

File: frontend/src/types/eval.ts
export interface EvalRun {
  id: string;
  run_id: string;
  strategy: string;
  reranking_enabled: boolean;
  faithfulness: number | null;
  answer_relevancy: number | null;
  context_precision: number | null;
  context_recall: number | null;
  per_question_results: Record<string, unknown> | null;
  created_at: string;
}

export interface EvalResults {
  results: EvalRun[];
  total: number;
}

export interface ABComparisonResult {
  metric: string;
  hybrid: number;
  dense_only: number;
  improvement: number;
}

export interface RerankComparisonResult {
  metric: string;
  with_reranking: number;
  without_reranking: number;
  improvement: number;
}
